Staking page update

Sirmorrison opened the pull request:
fix #660 
fix #674
Fix #648
* This PR implements the new staking page mock up UI.
* Redirects to ticket details page
* group overview, wallet page (including its sub pages), transaction (transaction details) page into sub packages in order to implement the redirect to tx details page functionality.
*** Further clean up of the overview, wallet, transaction packages will commence in subsequent PRs. 
* code clean up

Sirmorrison committed:
rename tickets package to staking

Sirmorrison committed:
rename ticket icons
- set stake button min width
- rename all instance of tickets to staking on overview page
- add No live stake label to live stake section
- hide see all btn when no live stake
- set see all btn size to mockup design
- add see all btn to stake record section
- hde see all btn when not stake record

Sirmorrison committed:
implement spend from unmixed account

Sirmorrison committed:
Clean up modal layout
- update staking modal to ui
- add ticket increment and decrement funtionality.
- comment out ticket review modal

Sirmorrison committed:
fix vsp selector bug when new vsp is selected
- use clickableList for vsp selector modal
- minimize vsp modal on background click.
- fix wallet total account balance
- automatically set last selected vsp host on resume
- clean up code

Sirmorrison committed:
Add success notification modal
- Add content alignment to info modal
- Add loader icon to ticket purchase modal
- Hide cancel button when loader icon is displayed.
- Remove password editor error when there is a change event.
- remove purchase overview modal file

Sirmorrison committed:
use ticket list view on ticket overview page

Sirmorrison committed:
Add tooltips to ticket list view
- update list view to mockup ui
- clean up code

Sirmorrison committed:
Add staking redirect to tx details page functionality
- group overview, transaction, wallet(subpages also) into package

Sirmorrison committed:
fix lint

Sirmorrison committed:
remove commented code

dreacot commented on ui/page/staking/overview.go:
```diff
@@ -36,17 +39,17 @@ type Page struct {
 
 	stakingOverview *dcrlibwallet.StakingOverview
-	liveTickets     []*transactionItem
+	liveStakes      []*transactionItem
```
```suggestion
	liveTickets      []*transactionItem
```
I believe live tickets is still very appropriate here, the only reason the button title was changed to stake was because you don't really purchase tickets if the money is returned back to you

dreacot commented on ui/page/staking/overview.go:
```diff
@@ -143,12 +146,12 @@ func (pg *Page) Layout(gtx layout.Context) layout.Dimensions {
 	widgets := []layout.Widget{
 		func(ctx layout.Context) layout.Dimensions {
 			return components.UniformHorizontalPadding(gtx, func(gtx layout.Context) layout.Dimensions {
-				return pg.ticketPriceSection(gtx)
```
same applies here with using tickets

dreacot commented on ui/page/staking/overview.go:
```diff
@@ -299,17 +315,21 @@ func (pg *Page) stakingRecordSection(gtx C) D {
 			}),
 			layout.Rigid(func(gtx C) D {
 				wdgs := []layout.Widget{
-					pg.stakingRecordIconCount(pg.Icons.TicketUnminedIcon, pg.stakingOverview.Unmined, "Unmined"),
-					pg.stakingRecordIconCount(pg.Icons.TicketImmatureIcon, pg.stakingOverview.Immature, "Immature"),
-					pg.stakingRecordIconCount(pg.Icons.TicketLiveIcon, pg.stakingOverview.Live, "Live"),
-					pg.stakingRecordIconCount(pg.Icons.TicketVotedIcon, pg.stakingOverview.Voted, "Voted"),
-					pg.stakingRecordIconCount(pg.Icons.TicketExpiredIcon, pg.stakingOverview.Expired, "Expired"),
-					pg.stakingRecordIconCount(pg.Icons.TicketRevokedIcon, pg.stakingOverview.Revoked, "Revoked"),
```
same applies here with using tickets

dreacot reviewed (changes requested):
1. Touching the backdrop should close the modal
<img width="795" alt="Screenshot 2021-11-04 at 5 14 21 PM" src="https://user-images.githubusercontent.com/25265396/140377124-b05f9e62-e7cc-4534-99f9-942a8708b6e4.png">

2. the got it text on the info dialogs should be by the right
<img width="437" alt="Screenshot 2021-11-04 at 5 14 46 PM" src="https://user-images.githubusercontent.com/25265396/140377212-248c01c7-fb16-40e0-8de6-ff821e20f928.png">

3. going to the all tickets page, switching wallets using the dropdown filter and then clicking on a ticket opens the ticket details page, but clicking the back arrow resets the filter to the default wallet on the dropdown, it should remain as the wallet you selected

<img width="799" alt="Screenshot 2021-11-04 at 5 15 37 PM" src="https://user-images.githubusercontent.com/25265396/140377365-5608abdf-1de8-4e2e-83f4-eeaa86db70d0.png">

4. Ideally the only thing that should have changed to stake was the button label on the modal to stake tickets, and the reason for that was because the users were not really purchasing tickets, and it was agreed to used the word stake there

Sirmorrison committed:
revert rename changes
- Minimise purchase modal when backdrop is clicked.
- Fix info modal button alignment.

dreacot reviewed (changes requested):
There should be a recent activity card on the staking page
<img width="416" alt="Screenshot 2021-11-08 at 6 56 33 PM" src="https://user-images.githubusercontent.com/25265396/140793434-0a61df15-bc98-48b7-a02d-58b0e9febb86.png">
